crates/rust-analyzer/src/flycheck.rs

```diff
@@ -401,7 +396,7 @@ impl FlycheckActor {
                             package_id = package_id.as_ref().map(|it| &it.repr),
                             "diagnostic received"
                         );
-                        self.diagnostics_received = DiagnosticsReceived::Yes;
```

Comment:
Oh, this line is wrong, that should be
```rust
if self.diagnostics_received == DiagnosticsReceived::No {
    self.diagnostics_received = DiagnosticsReceived::Yes;
}
```

Reply:
If we want to preserve the previous behavior, I believe this line should actually be:

```rust
                        self.diagnostics_received |= DiagnosticsReceived::Yes;
```

Where:

```rust
DiagnosticsReceived::Yes | DiagnosticsReceived::YesAndClearedForAll == DiagnosticsReceived::YesAndClearedForAll
DiagnosticsReceived::No  | DiagnosticsReceived::Yes == DiagnosticsReceived::Yes
DiagnosticsReceived::No  | DiagnosticsReceived::YesAndClearedForAll == DiagnosticsReceived::YesAndClearedForAll
```
 
 WDYT?